js/plugins/iq.js

```diff
@@ -0,0 +1,352 @@
+   * @public
+   * @method IqPlugin#init
+   */
+  this.init = function()
+  {
+    var missedEvents;
+    //if you need to process missed events, here is an example
+    if (_framework && OO._.isFunction(_framework.getRecordedEvents))
+    {
+      missedEvents = _framework.getRecordedEvents();
+    }
+    //use recorded events.
+
+    if (this.testMode)
+    {
+      trySetupAnalytics();
+    }
+    else if (!window.Ooyala)
+    {
+      OO.loadScriptOnce("https://analytics.ooyala.com/static/v3/analytics.js", trySetupAnalytics, sdkLoadError, SDK_LOAD_TIMEOUT);
+    }
+  };
+
+  /**
+   * [Required Function] Set the metadata for this plugin.
+   * @public
+   * @method IqPlugin#setMetadata
+   * @param  {object} metadata The metadata for this plugin
+   */
+  this.setMetadata = function(metadata)
+  {
+      OO.log( "Analytics Template: PluginID \'" + id + "\' received this metadata:", metadata);
+  };
+
+  /**
+   * [Required Function] Process an event from the Analytics Framework, with the given parameters.
+   * @public
+   * @method IqPlugin#processEvent
+   * @param  {string} eventName Name of the event
+   * @param  {Array} params     Array of parameters sent with the event
+   */
+  this.processEvent = function(eventName, params)
+  {
+    OO.log( "Analytics Template: PluginID \'" + id + "\' received this event \'" + eventName + "\' with these params:", params);
+    switch(eventName)
+    {
+      //OO.EVENTS.AUTHORIZATION_FETCHED -> OO.Analytics.STREAM_TYPE_UPDATED
+      case OO.Analytics.EVENTS.STREAM_TYPE_UPDATED:
+        if (params && params[0])
+        {
+          //Retrieve the stream type here.
+          //Possible values include OO.Analytics.STREAM_TYPE.VOD and OO.Analytics.STREAM_TYPE.LIVE_STREAM
+          var streamType = params[0].streamType;
+        }
+        break;
+      //OO.EVENTS.CONTENT_TREE_FETCHED -> OO.Analytics.EVENTS.VIDEO_CONTENT_METADATA_UPDATED.
+      case OO.Analytics.EVENTS.VIDEO_CONTENT_METADATA_UPDATED:
+        if (params && params[0])
+        {
+          duration = params[0].duration;
+          if (this.ooyalaReporter)
+          {
+            this.ooyalaReporter.initializeMedia(currentEmbedCode, contentType);
+            OO.log("IQ: Reported: initializeMedia() with args: " + currentEmbedCode + ", " + contentType);
+            this.ooyalaReporter.setMediaDuration(duration);
+            OO.log("IQ: Reported: setMediaDuration() with args: " + duration);
+          }
+          else
+          {
+            OO.log("Tried reporting event: " + OO.Analytics.EVENTS.VIDEO_CONTENT_METADATA_UPDATED +
+                   " but ooyalaReporter is: " + this.ooyalaReporter);
+          }
+        }
+        break;
+      //OO.EVENTS.EMBED_CODE_CHANGED -> OO.Analytics.EVENTS.VIDEO_SOURCE_CHANGED.
+      case OO.Analytics.EVENTS.VIDEO_SOURCE_CHANGED:
+        if (params && params[0] && params[0].metadata)
+        {
+          autoPlay = params[0].metadata.autoPlay;
```

Comment:
Does this track if the autoplay page parameter was true or if we actually autoplayed? For example, autoplay is currently disabled on mobile, so even if autoplay was set to true at page level, we still do not autoplay. We will be enabling autoplay on mobile soon as an option. If we want to track if the content actually autoplayed, we need to make sure we fetch the proper value.

Reply:
This is also pre-existing code. I just traced the call to the publication of the EMBED_CODE_CHANGED event that generates  OO.Analytics.EVENTS.VIDEO_SOURCE_CHANGED. It looks like "metadata" is just the page level params, so I think autoPlay here is just the page level option, not if anything was actually autoplayed or not.

Reply:
It might be better to remove this for now then if we don't get the correct autoplay option, so that when we do get it in, we won't have any bad data affecting queries. @pilievOoyala and @dallen42, what do you guys think? Might also be worth checking with Juan as well.

Reply:
I am fine commenting autoplay out, but if we do so we may also want to disable the handling of OO.Analytics.EVENTS.INITIAL_PLAYBACK_REQUESTED. reportPlayRequested() takes autoplay as an argument.